core/components/organisms/menu/Menu.tsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import { BaseProps } from '@/utils/types';
+import { Popover } from '@/index';
+import { PopoverProps } from '@/index.type';
+import { MenuGroup } from './MenuGroup';
+import { MenuItem } from './MenuItem';
+import { MenuList } from './MenuList';
+import { MenuTrigger } from './trigger/MenuTrigger';
+import SubMenu from './SubMenu';
+import classNames from 'classnames';
+import MenuContext from './MenuContext';
+import { focusListItem } from './trigger/utils';
+
+export interface MenuProps extends BaseProps {
+  /**
+   * Element to be rendered inside `Menu`
+   */
+  children: React.ReactNode;
+  /**
+   * Controls open/close of `Menu`
+   */
+  open?: boolean;
+  /**
+   * Defines position of `Menu`
+   */
+  position: PopoverProps['position'];
+  /**
+   * Defines trigger for the `Menu`
+   */
+  trigger?: React.ReactElement;
+  /**
+   * Specifies max height of `Menu`
+   */
+  maxHeight?: number;
+  /**
+   * Specifies min height of `Menu`
+   */
+  minHeight?: number;
+  /**
+   * Specifies width of `Menu`
+   */
+  width?: number;
+  /**
+   * Provide `ref` of the trigger element
+   */
+  triggerRef?: React.RefObject<HTMLDivElement>;
+  /**
+   * Describe unique id to the `Menu`
+   */
+  menuID?: string;
+  /**
+   * Defines coordinates where you need to position a popover
+   */
+  triggerCoordinates?: {
+    x: number;
+    y: number;
```

Comment:
Please clear it with design team whether to use triggerCoordinates in popever or not as it will be exposed on popover too.